sfm/ui/test_forms.py

```diff
@@ -163,3 +164,114 @@ def setUp(self):
+            "access_token_secret": "dummy_access_token_secret",
+            "date_added": "04/14/2016",
+        }
+
+    def test_form(self):
+        form = CredentialTwitterForm(self.data)
+        form.instance.user = self.user
+        self.assertTrue(form.is_valid())
+        credential = form.save()
+        self.assertTrue(credential.token,
+                        '{"consumer_key": "dummy_consumer_key",'
+                        '"consumer_secret": "dummy_consumer_secret",'
+                        '"access_token": "dummy_access_token",'
+                        '"access_token_secret": "dummy_access_token_secret"}')
+
+
+class CredentialWeiboFormTest(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.path = "/ui/credentials/flickr/create/"
+        self.user = User.objects.create_superuser(username="test_user",
+                                             email="test_user@test.com",
+                                             password="test_password")
+        self.data = {
+            "name": "test_weibo_credential",
+            "user": self.user.pk,
+            "platform": "weibo",
+            "api_key": "dummy_api_key",
+            "api_secret": "dummy_api_secret",
+            "redirect_uri": "dummy_redirect_uri",
+            "access_token": "dummy_access_token",
+            "date_added": "04/14/2016",
+        }
+
+    def test_form(self):
+        form = CredentialWeiboForm(self.data)
+        form.instance.user = self.user
+        self.assertTrue(form.is_valid())
+        credential = form.save()
+        self.assertTrue(credential.token,
+                        '{"api_key": "dummy_api_key",'
+                        '"api_secret": "dummy_api_secret",'
+                        '"redirect_uri": "dummy_redirect_uri",'
+                        '"access_token": "dummy_access_token"}')
+
+
+class CredentialUpdateFormTest(TestCase):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        user = User.objects.create_superuser(username="test_user",
+                                             email="test_user@test.com",
+                                             password="test_password")
+        self.credential = Credential.objects.create(user=user,
+                                                    name='Test Credential',
+                                                    platform='test platform',
+                                                    token='{"key1":"key1","key2":"key2"}')
+        self.path = '/ui/credentials/' + str(self.credential.pk) + '/update/'
+
+    def test_valid_data(self):
+        request = self.factory.get(self.path)
+        response = CredentialUpdateView.as_view()(request,
+                                                  pk=self.credential.pk)
+        form = CredentialForm({
+            'name': 'my test credential updated name',
+            'token': '{"key1":"updated_key1","key2","update_key2"}',
```

Comment:
Did you intend this to not be json? Not sure it has any effect on the test, but would be clearer.